### PR TITLE
Use sslmode require by default

### PR DESCRIPTION
**Pull request by pkoenig10**
See https://github.com/palantir/atlasdb/pull/5908#issuecomment-1039357029

### CODE REVIEW

File: atlasdb-ete-tests/docker/conf/atlasdb-ete.dbkvs.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ atlasdb:
       dbName: atlas
       dbLogin: palantir
       dbPassword: palantir
+      connectionParameters:
+        sslmode: disable
```
Comment (Jolyon-S) [inline atlasdb-ete-tests/docker/conf/atlasdb-ete.dbkvs.yml:25]: isn't this going to get overwritten? looks like you want a clause in the config overwriting if sslmode is explicitly set to disable.

Comment (pkoenig10) [reply atlasdb-ete-tests/docker/conf/atlasdb-ete.dbkvs.yml:25]: If it was overwritten the tests wouldn't work ;)

We apply the default `sslmode` before applying custom parameters. This ensure that consumers can still overwrite this if they want.

Comment (Jolyon-S) [reply atlasdb-ete-tests/docker/conf/atlasdb-ete.dbkvs.yml:25]: ugh I can't read, shouldn't be reviewing code at this time

of course, you're overriding it in the map

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/PostgresConnectionConfig.java [hunk 2]
```diff
@@ -79,14 +79,18 @@ public String getTestQuery() {
     @Value.Auxiliary
     public Properties getHikariProperties() {
         Properties props = new Properties();
-        // As per https://github.com/pgjdbc/pgjdbc/issues/1307, we need to replace ssl: true with sslmode: require
+
+        props.setProperty("sslmode", "require");
```
Comment (pkoenig10) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/PostgresConnectionConfig.java:83]: In the future we probably should change this to `verify-full`

**Comment by pkoenig10** [comment]
Given that we want `sslmode=verify-full` by default, I'm not sure we should push forward with this PR.

**Pull request closed without merging**